node/src/components/rpc_server.rs

```diff
@@ -352,56 +369,79 @@ where
                 deploy,
                 responder,
-            }) => self.handle_execute_deploy(effect_builder, block_header, *deploy, responder),
+            }) if self.speculative_exec_enable => {
+                self.handle_execute_deploy(effect_builder, block_header, *deploy, responder)
+            }
             Event::GetBlockResult {
                 maybe_id: _,
                 result,
                 main_responder,
-            } => main_responder.respond(*result).ignore(),
+            } if self.enable => main_responder.respond(*result).ignore(),
             Event::GetBlockTransfersResult {
                 result,
                 main_responder,
                 ..
-            } => main_responder.respond(*result).ignore(),
+            } if self.enable => main_responder.respond(*result).ignore(),
             Event::QueryGlobalStateResult {
                 result,
                 main_responder,
-            } => main_responder.respond(result).ignore(),
+            } if self.enable => main_responder.respond(result).ignore(),
             Event::QueryEraValidatorsResult {
                 result,
                 main_responder,
-            } => main_responder.respond(result).ignore(),
+            } if self.enable => main_responder.respond(result).ignore(),
             Event::GetBidsResult {
                 result,
                 main_responder,
-            } => main_responder.respond(result).ignore(),
+            } if self.enable => main_responder.respond(result).ignore(),
             Event::GetBalanceResult {
                 result,
                 main_responder,
-            } => main_responder.respond(result).ignore(),
+            } if self.enable => main_responder.respond(result).ignore(),
             Event::GetDeployResult {
                 hash: _,
                 result,
                 main_responder,
-            } => main_responder.respond(*result).ignore(),
+            } if self.enable => main_responder.respond(*result).ignore(),
             Event::GetPeersResult {
                 peers,
                 main_responder,
-            } => main_responder.respond(peers).ignore(),
+            } if self.enable => main_responder.respond(peers).ignore(),
+            Event::RpcRequest(RpcRequest::SpeculativeDeployExecute { .. })
+                if self.speculative_exec_enable =>
+            {
+                Effects::new()
+            }
+            Event::RpcRequest(_)
+            | Event::GetBalanceResult { .. }
+            | Event::GetBidsResult { .. }
+            | Event::GetBlockResult { .. }
+            | Event::GetBlockTransfersResult { .. }
+            | Event::GetDeployResult { .. }
+            | Event::GetPeersResult { .. }
+            | Event::QueryEraValidatorsResult { .. }
+            | Event::QueryGlobalStateResult { .. }
+                if !self.enable =>
+            {
+                Effects::new()
+            }
+            _ => unreachable!(),
```

Comment:
This will panic, correct? We should avoid panicking, even if we think this can't happen in bug-free software. I would change this to a `warn!` and log that we didn't expect to receive the request b/c endpoints should have been disabled.

Reply:
Done.